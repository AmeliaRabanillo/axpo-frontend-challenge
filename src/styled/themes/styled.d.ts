import 'styled-components';

declare module 'styled-components' {
    export interface DefaultTheme {
        fontFamily: string;
        fontSizes: {
            small: number;
            medium: number;
            large: number;
        };
        fontMeasureUnit: string;
        colors: {
            primary: string;
            black: string;
            white: string;
        };
    }
}