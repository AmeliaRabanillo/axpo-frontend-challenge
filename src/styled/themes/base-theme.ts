// @ts-ignore
import {DefaultTheme} from 'styled-components';

export const baseTheme: DefaultTheme = {
    fontFamily: '"Helvetica", sans-serif',
    fontSizes: {
        small: 20,
        medium: 32,
        large: 48,
    },
    fontMeasureUnit: 'px',

    colors: {
        primary: '#ff5d64',
        black: '#3d3d3d',
        white: '#f5e9d2',
    }
};
