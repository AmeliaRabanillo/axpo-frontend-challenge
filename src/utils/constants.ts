enum SearchTypeEnum {
    POP_DENSITY = 'population_density',
    DRONE_RES = 'drone_restriction',
}

const SEARCH_TYPE_DISPLAY_MAP = {
    [SearchTypeEnum.POP_DENSITY]: 'Population Density',
    [SearchTypeEnum.DRONE_RES]: 'Drone Restrictions'
};


enum RoutesEnum {
    START_PAGE = '/',
    POPULATION_DENSITY_TABLE_PAGE = '/population_density',
    DRONE_RESTRICTION_TABLE_PAGE = '/drone_restriction',
}

const MAP_SERVER_URL = 'https://api3.geo.admin.ch/rest/services/api/MapServer/identify';
const DRONE_RESTRICTIONS_LAYER = 'all:ch.bazl.einschraenkungen-drohnen';
const POPULATION_DENSITY_LAYER = 'all:ch.bfs.volkszaehlung-bevoelkerungsstatistik_einwohner';
const MAP_SERVER_GEOMETRY_TYPE = 'esriGeometryPoint';
const MAP_SERVER_SR = '4326';

export {
    SearchTypeEnum,
    RoutesEnum,
    SEARCH_TYPE_DISPLAY_MAP,
    MAP_SERVER_URL,
    DRONE_RESTRICTIONS_LAYER,
    POPULATION_DENSITY_LAYER,
    MAP_SERVER_GEOMETRY_TYPE,
    MAP_SERVER_SR,
}