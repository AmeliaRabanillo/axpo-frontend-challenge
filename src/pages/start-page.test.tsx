import React from 'react'
import 'jest-styled-components';
import StartPage from "./start-page";
import {ThemeProvider} from "styled-components";
import {baseTheme} from "../styled/themes/base-theme";
import {RoutesEnum, SEARCH_TYPE_DISPLAY_MAP} from "../utils/constants";
import {fireEvent, render} from "@testing-library/react";
import {screen} from '@testing-library/dom'

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
    useNavigate: () => mockedUsedNavigate,
}));

describe("Start Page tests", () => {
    afterEach(() => {
        mockedUsedNavigate.mockClear();
    })

    test("Start Page matches snapshot", () => {
        const {container} = (render(
            <ThemeProvider theme={baseTheme}>
                <StartPage/>
            </ThemeProvider>
        ));
        expect(container).toMatchSnapshot();
    });

    test("Start Page button redirects to density population table page", () => {
        const {container} = (render(
            <ThemeProvider theme={baseTheme}>
                <StartPage/>
            </ThemeProvider>
        ));

        const button = screen.getByText(SEARCH_TYPE_DISPLAY_MAP['population_density']);
        fireEvent.click(button);

        expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
        expect(mockedUsedNavigate).toHaveBeenCalledWith(RoutesEnum.POPULATION_DENSITY_TABLE_PAGE);
    });

    test("Start Page button redirects to drone restrictions table page", () => {
        const {container} = (render(
            <ThemeProvider theme={baseTheme}>
                <StartPage/>
            </ThemeProvider>
        ));

        const button = screen.getByText(SEARCH_TYPE_DISPLAY_MAP['drone_restriction']);
        fireEvent.click(button);

        expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
        expect(mockedUsedNavigate).toHaveBeenCalledWith(RoutesEnum.DRONE_RESTRICTION_TABLE_PAGE);
    });

});