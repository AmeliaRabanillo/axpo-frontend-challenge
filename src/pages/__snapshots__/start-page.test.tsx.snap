// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start Page tests Start Page matches snapshot 1`] = `
.c6 {
  color: #f5e9d2;
  background-color: #ff5d64;
  font-size: 16px;
  margin: 20px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 25px;
}

.c6 span {
  margin-right: 12px;
}

.c6:hover {
  cursor: pointer;
}

.c3 {
  color: #3d3d3d;
  font-size: 48px;
}

.c4 {
  color: #3d3d3d;
  font-size: 32px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 500px;
  min-width: 370px;
  -webkit-mask-repeat: no-repeat;
  -webkit-mask-position: center;
  -webkit-mask-image: url(https://s3-us-west-2.amazonaws.com/s.cdpn.io/3/sun.svg);
  -webkit-mask-image: url('/headerMaskMobile.svg');
  -webkit-mask-size: 100%;
  -webkit-mask-position: bottom;
}

@media (max-width:768px) {
  .c5 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c5 button {
    margin-left: 0;
    margin-right: 0;
  }
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media (max-width:768px) {
  .c1 {
    max-width: unset;
    width: 100%;
  }
}

@media (min-width:768px) {
  .c2 {
    margin-left: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <img
      class="c1"
      src="https://www.axpo.com/es/es/nuestros-clientes/_jcr_content/root/stage/stagev1/image1.coreimg.82.1280.png/1676642395487/0622-e-world-stage-wind-medium.png"
    />
    <div
      class="c2 main-container"
    >
      <h1
        class="c3"
      >
        Hello,
      </h1>
      <h2
        class="c4"
      >
        Select what you want to search
      </h2>
      <div
        class="c5"
      >
        <button
          class="c6"
        >
          <span>
            Population Density
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="GroupsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12.75c1.63 0 3.07.39 4.24.9 1.08.48 1.76 1.56 1.76 2.73V18H6v-1.61c0-1.18.68-2.26 1.76-2.73 1.17-.52 2.61-.91 4.24-.91zM4 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58C.48 14.9 0 15.62 0 16.43V18h4.5v-1.61c0-.83.23-1.61.63-2.29zM20 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85-.85-.37-1.79-.58-2.78-.58-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H24v-1.57zM12 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
            />
          </svg>
        </button>
        <button
          class="c6"
        >
          <span>
            Drone Restrictions
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MemoryIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 9H9v6h6V9zm-2 4h-2v-2h2v2zm8-2V9h-2V7c0-1.1-.9-2-2-2h-2V3h-2v2h-2V3H9v2H7c-1.1 0-2 .9-2 2v2H3v2h2v2H3v2h2v2c0 1.1.9 2 2 2h2v2h2v-2h2v2h2v-2h2c1.1 0 2-.9 2-2v-2h2v-2h-2v-2h2zm-4 6H7V7h10v10z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
